cmd/snap-repair/runner.go

```diff
@@ -69,14 +84,20 @@ var (
 	))
 )
 
-var ErrRepairNotFound = errors.New("repair not found")
+var (
+	ErrRepairNotFound    = errors.New("repair not found")
+	ErrRepairNotModified = errors.New("repair was not modified")
+)
 
 var (
 	maxRepairScriptSize = 24 * 1024 * 1024
 )
 
-// Fetch retrieves a stream with the repair with the given ids and any auxiliary assertions.
-func (run *Runner) Fetch(brandID, repairID string) (r []asserts.Assertion, err error) {
+// Fetch retrieves a stream with the repair with the given ids and any
+// auxiliary assertions. If revision>=0 Fetch sets in the request the
+// If-None-Match header with an E-Tag based on it, and
```

Comment:
Hmm, that's rather awkward wording (in particular the “sets in the request” bit). May I suggest `If revision >= 0 create an ETag based on the given revision, and set the request's If-None-Match header to this ETag.`? (also note ETag doesn't have a hyphen)